### PR TITLE
Add glean_usage ETL generation to generate_all_views

### PR DESCRIPTION
**Pull request by jklukas**
The new `generate_all_views` script is intended to replace `generate_views`
as the entrypoint for Jenkins. Its usage is demonstrated in the
`generate_and_publish_views` script.

This supports the move to user queries in the `mozdata` project.

### CODE REVIEW

File: script/generate_and_publish_views [hunk 1]
```diff
@@ -2,11 +2,32 @@
...
+esac
+done
+
+./script/generate_all_views --target-project "${TARGET_PROJECT}" --sql-dir "${SQL_DIR}"
+./script/publish_views --target-project "${TARGET_PROJECT}" "${SQL_DIR}"
+
+# We additionally make sure we have identical view definitions in moz-fx-data-derived-datasets and mozdata
+if [[ "$TARGET_PROJECT" == "moz-fx-data-shared-prod" ]]; then
+    ./script/publish_views --target-project moz-fx-data-derived-datasets "${SQL_DIR}"
+    ./script/publish_views --target-project mozdata "${SQL_DIR}"
```
Comment (scholtzan) [inline script/generate_and_publish_views:32]: When publishing to mozdata here should the `--user-facing-only` flag be added to ensure only user-facing views are published?

```suggestion
    ./script/publish_views --user-facing-only --target-project mozdata "${SQL_DIR}"
```

**Review by scholtzan — APPROVED** [review]

File: script/generate_and_publish_views (hunk 1 above)
Comment (whd) [reply script/generate_and_publish_views]: Any time we hard-code projects in bigquery-etl we lose the flexibility to support the [nonprod](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-pioneer/Jenkinsfile.bigquery#L37) environment.

https://github.com/mozilla/bigquery-etl/issues/1110

Comment (jklukas) [reply script/generate_and_publish_views]: Indeed. This `generate_and_publish_views` script is intended to demonstrate usage. My intention is that Jenkins would simply move its existing call to `generate_views` with `./script/generate_all_views --target-project ${project} --sql-dir /tmp/sql`.

This generate_and_publish_views logic is mimicing what Jenkins does. Only for shared-prod does it do the extra publishing to mozdata and derived-datasets. This block is skipped for nonprod where we don't want to republish.

**Review by whd — APPROVED** [review]
:+1:

**Pull request merged**